### PR TITLE
Dev/kk/book 4 implement oauth2

### PR DESCRIPTION
**Pull request by Sadymix**

### CODE REVIEW

File: src/main/java/com/pgs/booking/configuration/CustomAuthenticationProvider.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    public CustomAuthenticationProvider(UserService userDetailsService, PasswordEncoder passwordEncoder) {
+
+        PreAuthenticatedAuthenticationProvider preAuthenticatedAuthenticationProvider = new PreAuthenticatedAuthenticationProvider();
+        preAuthenticatedAuthenticationProvider.setPreAuthenticatedUserDetailsService(userDetailsService);
+
+        DaoAuthenticationProvider daoAuthenticationProvider = new DaoAuthenticationProvider();
+        daoAuthenticationProvider.setUserDetailsService(userDetailsService);
+        daoAuthenticationProvider.setPasswordEncoder(passwordEncoder);
+
+        authenticationProviders = ImmutableSet.of(preAuthenticatedAuthenticationProvider,daoAuthenticationProvider);
```
Comment (pszerszen) [inline src/main/java/com/pgs/booking/configuration/CustomAuthenticationProvider.java:34]: Use JDK native method instead of guava.

File: pom.xml [hunk 2]
```diff
@@ -88,30 +84,33 @@
...
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-jwt</artifactId>
+            <version>1.1.1.RELEASE</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.google.guava</groupId>
+            <artifactId>guava</artifactId>
+            <version>31.0.1-jre</version>
+        </dependency>
```
Comment (pszerszen) [inline pom.xml:108]: I don't thing it's necessary to have that. Most of the utils from guava are more or less in JDK now.

File: pom.xml [hunk 3]
```diff
@@ -88,30 +84,33 @@
...
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
```
Comment (pszerszen) [inline pom.xml:113]: do we really need junit 4?

File: src/test/java/com/pgs/booking/controller/PassengerControllerTest.java [hunk 4]
```diff
@@ -22,21 +22,18 @@
...
 import static org.mockito.Mockito.verify;
-import static org.mockito.Mockito.when;
-import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
-import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
-@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+@RunWith(SpringRunner.class)
```
Comment (pszerszen) [inline src/test/java/com/pgs/booking/controller/PassengerControllerTest.java:29]: redundant

File: src/test/java/com/pgs/booking/controller/PassengerControllerTest.java [hunk 5]
```diff
@@ -98,56 +85,4 @@ void testGetSinglePassenger() throws Exception {
...
-                        .accept(MediaType.APPLICATION_JSON))
-                .andExpect(status().isOk())
-                .andExpect(jsonPath("$.id").exists())
-                .andExpect(jsonPath("$.firstName").exists())
-                .andExpect(jsonPath("$.lastName").exists())
-                .andExpect(jsonPath("$.email").exists())
-                .andExpect(jsonPath("$.country").exists())
-                .andExpect(jsonPath("$.telephone").exists());
-        verify(passengerService).addPassenger(CREATE_UPDATE_PASSENGER_DTO);
-    }
```
Comment (pszerszen) [inline src/test/java/com/pgs/booking/controller/PassengerControllerTest.java:118]: bring back and adjust to your changes

File: src/test/java/com/pgs/booking/controller/PassengerControllerTest.java [hunk 6]
```diff
@@ -98,56 +85,4 @@ void testGetSinglePassenger() throws Exception {
...
-                .accept(MediaType.APPLICATION_JSON))
-                .andExpect(status().isOk())
-                .andExpect(MockMvcResultMatchers.jsonPath("$.id").value(id))
-                .andExpect(MockMvcResultMatchers.jsonPath("$.firstName").value(PASSENGER_DTO.getFirstName()))
-                .andExpect(MockMvcResultMatchers.jsonPath("$.lastName").value(PASSENGER_DTO.getLastName()))
-                .andExpect(MockMvcResultMatchers.jsonPath("$.email").value(PASSENGER_DTO.getEmail()))
-                .andExpect(MockMvcResultMatchers.jsonPath("$.country").value(PASSENGER_DTO.getCountry()))
-                .andExpect(MockMvcResultMatchers.jsonPath("$.telephone").value(PASSENGER_DTO.getTelephone()));
-        verify(passengerService).editPassenger(id, CREATE_UPDATE_PASSENGER_DTO);
-    }
```
Comment (pszerszen) [inline src/test/java/com/pgs/booking/controller/PassengerControllerTest.java:137]: bring back and adjust to your changes

**Review by pszerszen — CHANGES_REQUESTED** [review]

File: src/test/java/com/pgs/booking/controller/PassengerControllerTest.java (hunk 4 above)
Comment (Sadymix) [reply src/test/java/com/pgs/booking/controller/PassengerControllerTest.java:32]: @RunWith(SpringRunner.class) is redundant. We need @SpringBootTest for WebApplicationContext

File: src/test/java/com/pgs/booking/controller/PassengerControllerTest.java [hunk 7]
```diff
@@ -62,14 +59,12 @@
     public void setUP() {
         mockMvc = MockMvcBuilders
                 .webAppContextSetup(wac)
-                .apply(springSecurity())
                 .build();
     }
 
-    @WithMockUser
+    @SneakyThrows
```
Comment (pszerszen) [inline src/test/java/com/pgs/booking/controller/PassengerControllerTest.java:65]: Some methods use @SneakyThrows and some just have throws declaration. Pick one approach and stick to it.

**Review by pszerszen — APPROVED** [review]
Approvable, but I left one minor cosmetic comment.

**Review by pszerszen — APPROVED** [review]

**Pull request merged**